articles/azure-functions/durable/durable-functions-checkpointing-and-replay.md

```diff
@@ -124,8 +124,16 @@ The replay behavior creates constraints on the type of code that can be written
 
   If orchestrator code needs to get the current date/time, it should use the [CurrentUtcDateTime](https://azure.github.io/azure-functions-durable-extension/api/Microsoft.Azure.WebJobs.DurableOrchestrationContext.html#Microsoft_Azure_WebJobs_DurableOrchestrationContext_CurrentUtcDateTime) (.NET) or `currentUtcDateTime` (JavaScript) API, which is safe for replay.
 
-  If orchestrator code needs to generate a random GUID, it should use the [NewGuid](https://azure.github.io/azure-functions-durable-extension/api/Microsoft.Azure.WebJobs.DurableOrchestrationContext.html#Microsoft_Azure_WebJobs_DurableOrchestrationContext_NewGuid) (.NET) API, which is safe for replay, or delegate GUID generation to an activity function (JavaScript), as in this example:
+  If orchestrator code needs to generate a random GUID, it should use the [NewGuid](https://azure.github.io/azure-functions-durable-extension/api/Microsoft.Azure.WebJobs.DurableOrchestrationContext.html#Microsoft_Azure_WebJobs_DurableOrchestrationContext_NewGuid) (.NET) API, which is safe for replay, or delegate GUID generation to an activity function, as in these examples:
 
+  ### C#
+  
+  ```csharp
+  public static Guid GenGuid([ActivityTrigger] DurableActivityContext context) => Guid.NewGuid();
```

Comment:
Since the recommendation is to use the `NewGuid` API, I think it's a bit misleading for us to show a code sample of a less performant alternative. In the case of JavaScript, we've actually added support for `newGuid()` since the [v1.2.0](https://github.com/Azure/azure-functions-durable-js/releases/tag/1.2.0) release so I'm wondering if we should remove the activity function suggestion altogether.

Reply:
gotchya - do you want to have some other sort of example here, then?

Reply:
I would say leave out the example and just mention the `NewGuid()` (.NET) and `newGuid()` (JS) APIs as the workarounds in the above text.